src/cartesian/CartesianGrid.js

```diff
@@ -131,6 +136,81 @@ class CartesianGrid extends Component {
     return <g className="recharts-cartesian-grid-vertical">{items}</g>;
   }
 
+  /**
+   * Draw vertical grid stripes filled by colors
+   * @param {Array} verticalPoints either passed in as props or generated from function
+   * @return {Group} Vertical stripes
+   */
+  renderVerticalStripes(verticalPoints) {
+    const { verticalFill } = this.props;
+    if (!verticalFill || !verticalFill.length) { return null; }
+
+    const { fillOpacityVertical, x, y, width, height } = this.props;
+    let verticalPointsUpdated = verticalPoints;
+    if (x === 0) {
+      verticalPointsUpdated.unshift(0);
+    }
+
+    const items = verticalPointsUpdated.map((entry, i) => {
+      let lineWidth = Math.max(0, (i === 0) ? verticalPointsUpdated[i+1]-x : verticalPointsUpdated[i+1] !== undefined ? verticalPointsUpdated[i+1] - entry : width - entry);      
```

Comment:
You need lint for this line.
And you should simplify this line.
```js
const lineWidth = !_.isNil(verticalPointsUpdated[i + 1] ? verticalPointsUpdated[i + 1] - entry :  (x + width - entry);

if (lineWidth <= 0) { return null; } 
```

Reply:
ok, I will do, thanks.